executor/distsql.go

```diff
@@ -527,6 +527,13 @@ func (e *IndexLookUpExecutor) startIndexWorker(ctx context.Context, workCh chan<
 			SetFromInfoSchema(infoschema.GetInfoSchema(e.ctx))
 
 		for partTblIdx, kvRange := range kvRanges {
+			// check if executor is closed
+			select {
+			case <-e.finished:
+				break
```

Comment:
maybe this doesn't take effect, `break` only break `select`, will kvRange still will access

Reply:
Nice catch... It is fixed now. PTAL at your convenience.